example/v1/types_stable.go

```diff
@@ -0,0 +1,67 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=1
+
+// StableConfigType is a stable config type that may include TechPreviewNoUpgrade fields.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type StableConfigType struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the StableConfigType.
+	Spec StableConfigTypeSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the StableConfigType.
+	Status StableConfigTypeStatus `json:"status,omitempty"`
+}
+
+// StableConfigTypeSpec is the desired state
+type StableConfigTypeSpec struct {
+	// coolNewField is a field that is for tech preview only.  On normal clusters this shouldn't be present
+	//
+	// +kubebuilder:validation:Required
```

Comment:
Nit: You can't add a required field to an existing API unless it's defaulted right? Else consumers won't have a value here on upgrade